src/preload.py

```diff
@@ -79,6 +79,33 @@ def prepare_global_partitions():
         return result
     return get_partitions(IMAGE)
 
+class RetryCounter:
+    def __init__(self):
+        self.counter = {}
+
+    @staticmethod
+    def key(*args):
+        return " ".join(str(e) for e in args)
+
+    def clear(self, *args):
+        del self.counter[self.key(*args)]
+
+    def inc(self, *args):
+        key = self.key(*args)
+        self.counter[key] = self.counter.setdefault(key, 0) + 1
+        return self.counter[key]
+
+retry_counter = RetryCounter()
+
+def losetup_wrap(*args, **kwargs):
+    count = retry_counter.inc(*args)
+    if count > 1:
+        msg = """
+Retrying (count={}) losetup {}
+Hint: if using a Virtual Machine, consider increasing the CPU count.
```

Comment:
I've learned this hint the hard way, :-) using a VirtualBox VM with Debian, on a macOS host. With the default CPU count of 1, `losetup` would fail numerous times, even though the the overall laptop resource usage was relatively low (free physical memory, VM not swapping, VM CPU between 50% and 100%). Increasing the CPU count to 2 reduces the number of losetup retries a lot, and increasing it to 4 makes losetup succeed without any retries (almost always, occasionally a single retry).